### ISSUE TITLE
audio related crash and exception

### ISSUE THREAD

**Issue by l3lackShark** [ppy/osu#11029]
**Describe the crash:** 
First crash occured on 2020.1121.0.0, second on 2020.1128.0.0 (the game was updating in the meantime) both at retry attempts of [this](https://osu.ppy.sh/beatmapsets/1265600#taiko/2634936) beatmap. The exceptions seem to be different.  Second crash was soft, It just said that error was reported and I did not had any audio except game sounds ever since. Worth noting that I could not reproduce any of these anymore. 

**osu!lazer version:** 
2020.1121.0.0 and 2020.1128.0.0

**Logs:**
First:
```
2020-12-01 06:11:58 [verbose]: Unhandled exception has been allowed with 0 more allowable exceptions .
2020-12-01 06:11:58 [error]: An unhandled error has occurred.
2020-12-01 06:11:58 [error]: System.ObjectDisposedException: Can not play disposed samples.
2020-12-01 06:11:58 [error]: Object name: 'osu.Framework.Audio.Sample.SampleChannelBass'.
2020-12-01 06:11:58 [error]: at osu.Framework.Audio.Sample.SampleChannel.Play(Boolean restart)
2020-12-01 06:11:58 [error]: at osu.Framework.Audio.Sample.SampleChannelBass.Play(Boolean restart)
2020-12-01 06:11:58 [error]: at osu.Framework.Graphics.Audio.DrawableSample.Play(Boolean restart)
2020-12-01 06:11:58 [error]: at osu.Game.Skinning.SkinnableSound.<Play>b__18_0(DrawableSample c)
2020-12-01 06:11:58 [error]: at osu.Framework.Extensions.IEnumerableExtensions.EnumerableExtensions.ForEach[T](IEnumerable`1 collection, Action`1 action)
2020-12-01 06:11:58 [error]: at osu.Game.Skinning.SkinnableSound.Play()
```

Second:
```
2020-12-01 06:14:46 [verbose]: Unhandled exception has been allowed with 0 more allowable exceptions .
2020-12-01 06:14:46 [error]: An unobserved error has occurred.
2020-12-01 06:14:46 [error]: System.AggregateException: A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. As a result, the unobserved exception was rethrown by the finalizer thread. (Collection was modified; enumeration operation may not execute.)
2020-12-01 06:14:46 [error]: ---> System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
2020-12-01 06:14:46 [error]: at System.Collections.Generic.List`1.Enumerator.MoveNextRare()
2020-12-01 06:14:46 [error]: at osu.Framework.Audio.AudioCollectionManager`1.Dispose()
2020-12-01 06:14:46 [error]: at osu.Game.Rulesets.UI.DrawableRulesetDependencies.FallbackSampleStore.Dispose()
2020-12-01 06:14:46 [error]: at osu.Game.Rulesets.UI.DrawableRulesetDependencies.Dispose(Boolean disposing)
2020-12-01 06:14:46 [error]: at osu.Game.Rulesets.UI.DrawableRulesetDependencies.Dispose()
```

Full Logs:
[runtime.log](https://github.com/ppy/osu/files/5620624/runtime.log)
[runtime2.log](https://github.com/ppy/osu/files/5620625/runtime2.log)

**Computer Specifications:** 
1070ti
Ryzen 3700x
16GB RAM

**Comment by peppy** [comment]
First is a duplicate of https://github.com/ppy/osu/issues/5798.

**Comment by peppy** [comment]
Second has been resolved, first is duplicate. Closing.
